### ISSUE TITLE
Bulk insert with with include option.

### ISSUE THREAD

**Issue by SargeKhan** [sequelize/sequelize#5498]
saving an object with TableName.create({data}, {include: OtherTable}) works, but when using TableName.bulkCreate(ArrayData, {include:OtherTable}) creates entry for TableName but OtherTable value is set to null. Is there a problem with bulk insert?

**Comment by janmeier** [comment]
There's not a problem per se, its just not implemented :)

**Comment by mickhansen** [comment]
This would only ever work by converting bulkCreate into a loop of create internally.

**Comment by ryanouyang** [comment]
you can do it manually.
```javascript
// for example
ModelA.belongsTo(ModelB);
// let's say ModelA use freeze table name 'model_a', ModelB use freeze table name with 'model_b', and underscored to true by default when define these models.
// when you add record, you add the references manually
var model_a_records = [];
model_a_records.push({
	modelBId: modelBInstance.id,
	model_b: modelBInstance
	// model_a_field: ...
});
// then you are good to use bulk create without include option.
ModelA.bulkCreate(mode_a_records)
```

**Comment by illumimari** [comment]
+1

**Comment by TangMonk** [comment]
+1

**Comment by KevinGruber** [comment]
@ryanouyang can you please clearify that a lottli more? i can't get that to work. 
Also is being worked on that feature?

**Comment by aaewong** [comment]
@KevinGruber For two tables A and B I have the relation `A.hasMany(B)`. To bulk insert to A and include nested B records I do something like this:

```javaScript
const aRecords = []; // some objects for table A including the nested B objects
A.bulkCreate(aRecords, { 
  individualHooks: true // to return the ids of created rows
}).then((createdInstances) => {
  // Extract the nested B objects and link to the corresponding ID of the created instance
 const bRecords = linkToA(aRecords, createdInstances)
 return B.bulkCreate(bRecordsWithAId);
})
```

**Comment by mavidser** [comment]
+1

**Comment by rashmi-nyllabs** [comment]
+1

**Comment by framled** [comment]
+1

**Comment by Deilan** [comment]
It is an issue! Moreover, it's still highly requested! :)

**Comment by SirWojtek** [comment]
Similar problem: #3277

**Comment by sonrisacc** [comment]
@aaewong Hi, would you mind elaborate on the linkToA function? I set individualHooks to true, but it didn't emit the  the ids of created rows, though I can still access it through createdInstances...but then I feel like I'm missing sth to include  bRecords with the corresponding ArecordsID... pretty new with sequelize so plz bear with me... Thank you so much!!!

**Comment by aaewong** [comment]
@sonrisacc Since ```bulkCreate``` doesn't create nested records (```bRecords``` in my example), we need to extract the nested objects from the parent (```aRecords```) and give them the id of their parent. ```linkToA``` returns a list of ```bRecords``` including the corresponding id from the ```createdInstances``` from the ```bulkCreate``` on table A. 

Example:
```javaScript
const linkToA = (aRecords, createdInstances) => {
  const bRecords = createdInstances.map((instance, index) => {
    aRecords[index].bRecords.map((res) => Object.assign({ aId: instance.id }, res))
  })
  return [].concat(...bRecords)
}
```

**Comment by sushantdhiman** [comment]
Duplicate https://github.com/sequelize/sequelize/issues/3277

**Comment by aqdas95** [comment]
> @KevinGruber For two tables A and B I have the relation `A.hasMany(B)`. To bulk insert to A and include nested B records I do something like this:
> 
> ```js
> const aRecords = [] // some objects for table A including the nested B objects
> A.bulkCreate(aRecords, { 
>   individualHooks: true // to return the ids of created rows
> }).then((createdInstances) => {
>   // Extract the nested B objects and link to the corresponding ID of the created instance
>  const bRecords = linkToA(aRecords, createdInstances)
>  return B.bulkCreate(bRecordsWithAId)
> })
> ```

@aaewong can you please quote an example for it too?

**Comment by mohit26chauhan** [comment]
Any update on this feature request. Is this feature is planned to be part of next releases? Should we wait for it?

**Comment by adrianodennanni** [comment]
It's solved (v5.14.0, https://github.com/sequelize/sequelize/issues/3277)
